client/src/commands/run.ts

```diff
@@ -204,14 +215,39 @@ export async function runTask(
     setIsExecutingCode(false);
     commands.executeCommand("setContext", "SAS.running", false);
   });
-  session.onExecutionLogFn = onExecutionLog ?? appendExecutionLogFn;
-  session.onSessionLogFn = onSessionLog ?? appendSessionLogFn;
+
+  session.onExecutionLogFn = sasDiagnostic.generateLogFn(
+    codeDoc,
+    appendExecutionLogFn,
+  );
+  session.onSessionLogFn = appendSessionLogFn;
 
   messageEmitter.fire(`${l10n.t("Connecting to SAS session...")}\r\n`);
   !cancelled && (await session.setup(true));
 
   messageEmitter.fire(`${l10n.t("SAS code running...")}\r\n`);
-  return cancelled ? undefined : session.run(code);
+  return cancelled
+    ? undefined
+    : session.run(codeDoc.getWrappedCode()).then((results) => {
+        showRunResult(results, messageEmitter, taskLabel);
+      });
+}
+
+function showRunResult(
```

Comment:
I would suggest to keep `showRunResult` in `SasTasks.ts` as before. It's task-specific.